integrations/terraform-cloud/main.py

```diff
@@ -54,6 +55,29 @@ async def enrich_state_versions_with_output_data(
         return enriched_state_versions
 
 
+async def enrich_workspaces_with_tags(
+    http_client: TerraformClient, workspaces: List[dict[str, Any]]
+) -> list[dict[str, Any]]:
+    async def get_tags_for_workspace(workspace: dict[str, Any]) -> dict[str, Any]:
+        tags = []
+        async for tag_batch in http_client.get_workspace_tags(workspace["id"]):
+            tags.extend(tag_batch)
+        return {**workspace, "__tags": tags}
+
+    tasks = [get_tags_for_workspace(workspace) for workspace in workspaces]
+    enriched_workspaces = [await task for task in asyncio.as_completed(tasks)]
+
+    return enriched_workspaces
+
```

Comment:
This might cause rate limit, I would suggest adding a semaphore.

minor thought, maybe we wouldn't want to fail the entire sync if we were unable to get tags for one of the orgs, worth checking if maybe we can be resilient to errors like 404 / 403 / 401 and just continue.